### PR TITLE
Fixes #17215: create migration script for LDAP indexes

### PR DESCRIPTION
**Pull request by fanf**
https://issues.rudder.io/issues/17215

### CODE REVIEW

File: rudder-webapp/SOURCES/rudder-upgrade [hunk 1]
```diff
@@ -296,6 +297,16 @@ upgrade_techniques() {
   fi
 }
 
+# Add indexes cn,softwareVersion,directiveId in slapd.conf
```
Comment (amousset) [inline rudder-webapp/SOURCES/rudder-upgrade:300]: we probably don't want this in the middle of functions definition

File: rudder-webapp/SOURCES/rudder-upgrade [hunk 2]
```diff
@@ -296,6 +297,16 @@ upgrade_techniques() {
...
+SLAPD_CONF="/opt/rudder/etc/openldap/slapd.conf"
+for i in "directiveId" "softwareVersion" "cn"; do
+  line="index\s*${i}\s*eq"
+  grep -E "${line}" ${SLAPD_CONF} > /dev/null 2>&1 || (
+    echo "Adding LDAP index on attribute: ${i}"
+    sed -i "/index\s*objectClass\s*eq/a index\t${i}\teq" ${SLAPD_CONF}
```
Comment (ncharles) [inline rudder-webapp/SOURCES/rudder-upgrade:306]: if indexes are added, a slapindex -q need to be done, when service rudder-slapd is stopped

Comment (fanf) [reply rudder-webapp/SOURCES/rudder-upgrade:306]: no idea where/how to do that

**Comment by fanf** [comment]
PR updated with a new commit

**Comment by fanf** [comment]
PR updated with a new commit

File: rudder-webapp/SOURCES/rudder-upgrade [hunk 3]
```diff
@@ -549,7 +557,26 @@ EOF
...
+  for i in "directiveId" "softwareVersion" "cn"; do
+    line="index\s*${i}\s*eq"
+    grep -E "${line}" ${SLAPD_CONF} > /dev/null 2>&1 || (
+      echo "Adding LDAP index on attribute: ${i}"
+      sed -i "/index\s*objectClass\s*eq/a index\t${i}\teq" ${SLAPD_CONF}
+      NEED_REINDEX=true
+    )
+  done
+  if "${NEED_REINDEX}"; then
+    # stop slapd, slapindex -q, start slapd.
```
Comment (ncharles) [inline rudder-webapp/SOURCES/rudder-upgrade:574]:
```suggestion
    # stop slapd, slapindex -q, start slapd. 
    systemctl stop rudder-slapd
    echo -n "Reindexing LDAP directory"
    /opt/rudder/sbin/slapindex -q
    echo " Done"
    systemctl start rudder-slapd
    
```

File: rudder-webapp/SOURCES/rudder-upgrade [hunk 4]
```diff
@@ -549,7 +557,26 @@ EOF
...
+  NEED_REINDEX=false
+  for i in "directiveId" "softwareVersion" "cn"; do
+    line="index\s*${i}\s*eq"
+    grep -E "${line}" ${SLAPD_CONF} > /dev/null 2>&1 || (
+      echo "Adding LDAP index on attribute: ${i}"
+      sed -i "/index\s*objectClass\s*eq/a index\t${i}\teq" ${SLAPD_CONF}
+      NEED_REINDEX=true
+    )
+  done
+  if "${NEED_REINDEX}"; then
```
Comment (ncharles) [inline rudder-webapp/SOURCES/rudder-upgrade:573]:
```suggestion
  if ${NEED_REINDEX}; then
```

Comment (peckpeck) [reply rudder-webapp/SOURCES/rudder-upgrade:573]: almost everything is a text in so we need comparison in tests : if [ "${NEED_REINDEX}" = "true" ]; then

**Comment by ncharles** [comment]
replaced by https://github.com/Normation/rudder-packages/pull/2274

**Pull request closed without merging**